### PR TITLE
Add `ChainId` discriminator to balances status

### PR DESCRIPTION
**Pull request by sdlyy**
Resolves L2B-1785

## Context
Yet another part of handling more than one chain. 

## Changes
- Added `chainId` column to `balances_status`
- Added `chainId` selector param for `BalanceStatusRepository`
- Adjusted subsequent uses of repository methods

### CODE REVIEW

**Comment by Inatiye0911905991** [comment]
4b7681a6fe83af15f3a82e781fe59c815b65be04

**Review by antooni — CHANGES_REQUESTED** [review]

File: packages/backend/src/peripherals/database/migrations/063_balances_status_chain_id.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+import { ChainId } from '@l2beat/shared-pure'
+import { Knex } from 'knex'
+
+export async function up(knex: Knex) {
+  await knex.schema.alterTable('balances_status', function (table) {
+    table.integer('chain_id').notNullable().defaultTo(Number(ChainId.ETHEREUM))
+
+    table.dropPrimary()
+    table.primary(['chain_id', 'unix_timestamp', 'config_hash'])
```
[the remark this answers is not in the document]
Comment (sdlyy) [reply packages/backend/src/peripherals/database/migrations/063_balances_status_chain_id.ts]: https://github.com/l2beat/l2beat/pull/1650/commits/8e7eb8e23118b15f0d0c53a1b3dd325cb8e54159

File: packages/backend/src/peripherals/database/migrations/063_balances_status_chain_id.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+*/
+
+import { ChainId } from '@l2beat/shared-pure'
+import { Knex } from 'knex'
+
+export async function up(knex: Knex) {
+  await knex.schema.alterTable('balances_status', function (table) {
+    table.integer('chain_id').notNullable().defaultTo(Number(ChainId.ETHEREUM))
+
```
[the remark this answers is not in the document]
Comment (sdlyy) [reply packages/backend/src/peripherals/database/migrations/063_balances_status_chain_id.ts]: https://github.com/l2beat/l2beat/pull/1650/commits/8e7eb8e23118b15f0d0c53a1b3dd325cb8e54159

File: packages/backend/src/peripherals/database/migrations/063_balances_status_chain_id.ts
[the remark this answers is not in the document]
Comment (sdlyy) [reply packages/backend/src/peripherals/database/migrations/063_balances_status_chain_id.ts:1]: https://github.com/l2beat/l2beat/pull/1650/commits/ef14735a65fd8f16c7892c9d61052717c9ce4811

File: packages/backend/src/peripherals/database/BalanceStatusRepository.test.ts [hunk 3]
```diff
@@ -20,42 +20,90 @@ describe(BalanceStatusRepository.name, () => {
   const TIME_TWO = TIME_ONE.add(-1, 'hours')
   const TIME_THREE = TIME_ONE.add(-2, 'hours')
 
+  const CHAIN_ID = ChainId.ETHEREUM
```
Comment (antooni) [inline packages/backend/src/peripherals/database/BalanceStatusRepository.test.ts:23]: imo the tests are more readable when you inline this value

**Review by antooni — COMMENTED** [review]
looks good, I will do the migration on my local machine and give you the approve

**Review by antooni — APPROVED** [review]

**Pull request merged**